src/PhpBrew/VariantBuilder.php

```diff
@@ -315,34 +315,27 @@ public function __construct()
             return;
         };
 
-        $this->variants['curl'] = function (Build $build, $prefix = null) {
-            if ($prefix) {
-                return "--with-curl=$prefix";
-            }
-
-            if ($prefix = Utils::findIncludePrefix('curl/curl.h')) {
-                return "--with-curl=$prefix";
-            }
+        $this->variants['curl'] = function (Build $build, $val = null) {
+            $prefix = Utils::findPrefix(array(
+                new BrewPrefixFinder('curl'),
+                new PkgConfigPrefixFinder('libcurl'),
+                new IncludePrefixFinder('curl/curl.h'),
+            ));
 
-            if ($prefix = Utils::getPkgConfigPrefix('libcurl')) {
-                return "--with-curl=$prefix";
-            }
+            if ($build->compareVersion('7.4') < 0) {
+                if ($val) {
+                    return "--with-curl=$val";
+                }
 
-            if ($bin = Utils::findBin('brew')) {
-                if ($prefix = exec_line("$bin --prefix curl")) {
-                    if (file_exists($prefix)) {
-                        return "--with-curl=$prefix";
-                    }
-                    echo "homebrew prefix '$prefix' doesn't exist. you forgot to install?\n";
+                if ($prefix !== null) {
+                    return '--with-curl=' . $prefix;
                 }
-            }
-            if ($bin = Utils::findBin('curl-config')) {
-                if ($prefix = exec_line("$bin --prefix")) {
-                    if (file_exists($prefix)) {
-                        return "--with-curl=$prefix";
-                    }
-                    echo "homebrew prefix '$prefix' doesn't exist. you forgot to install?\n";
+            } else {
+                if ($prefix !== null) {
+                    $build->setPkgConfigPath($prefix.'/lib/pkgconfig');
                 }
+
+                return '--with-curl';
```

Comment:
This `return` is redundant. If it's omitted, the result will be the same.

Reply:
I'll remove it.